packages/desktop-client/src/components/reports/ReportSidebar.js

```diff
@@ -80,6 +82,15 @@ export function ReportSidebar({
     }
   };
 
+  const onChangeDatePaused = cond => {
+    setDatePaused(cond);
+    if (cond === 'live') {
+      onSelectRange(dateRange);
+    } else {
+      onChangeDates(startDate, endDate);
+    }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I don't quite understand why we have 2x different callbacks here. Why not unify them into a single one `onDateChange`/`onChangeDate`? Would make things easier and reduce duplication. 

Surely on the consumer level you're not storing 2x different dates for the custom reports, but instead a single date (or if you are storing 2x dates - that's actually a problem that we will need to solve before we save them to the DB).

Reply:
I think it was unesseccary extra code. I think it's better now.

For reference I am not saving 2 dates. Current saved fields are: (isDateStatic, dateRange, startDate, endDate). First one tells us if it's a static or live graph, if live then next one saves the dateRange for the live graph, if static the final 2 tell us the start/end date.

Concievably this could all be stored as one array if that's easier to maintain.

dateArray{isDateStatic: (true/false), dateRange: string, startDate: yyyy-mm-dd, endDate: yyyy-mm-dd}

Reply:
Those are 2x date-ranges though: 1) dateRange; 2) startDate + endDate. We should not be saving duplicate information in the DB (eventually).

You could combine them in just a single date object: startDate + endDate + isStatic. If `isStatic = false` then you can use `endDate - startDate` to calculate the dateRange value.

Reply:
How would that equation know the difference between "thisMonth" and "lastMonth"? Especially for a graph I save now then don't open again for a couple months? Seems overly complex for something that can be solved by saving it directly as a simple string.

Reply:
Good argument. I hadn't thought about the this month/last month problem. In that case it does make sense to keep all the values.